fbgemm_gpu/src/dram_kv_embedding_cache/fixed_block_pool.h

```diff
@@ -8,20 +8,122 @@
 
 #pragma once
 
+#include <chrono>
 #include <cstddef>
 #include <memory_resource>
+#include <numeric>
 #include <stdexcept>
 #include <vector>
 
 #include <cassert>
+#include <cmath>
 
 namespace kv_mem {
+static constexpr uint32_t kMaxInt31Counter = 2147483647;
+
 class FixedBlockPool : public std::pmr::memory_resource {
  public:
+  // Chunk metadata
+  struct ChunkInfo {
+    void* ptr;         // Memory block pointer
+    std::size_t size;  // Total size
+    std::size_t alignment;
+  };
+
+  // Metadata structure (publicly accessible)
+  // alignas(8) MetaHeader >= sizeof(void*), avoid mempool block too small.
+  struct alignas(8) MetaHeader {  // 16bytes
+    int64_t key;                  // feature key 8bytes
+    uint32_t timestamp;   // 4 bytes，the unit is second, uint32 indicates a
+                          // range of over 120 years
+    uint32_t count : 31;  // only 31 bit is used, max value is 2147483647
```

Comment:
wondering are we able to make the key and count as optional. in the inference side, memory is more restricted, and key/count won't be used anyway